lib/nerves_hub_link_http.ex

```diff
@@ -1,25 +1,54 @@
 defmodule NervesHubLinkHTTP do
+  @moduledoc """
+  Official client to use HTTP polling for firmware updates from NervesHub
+  """
+
   require Logger
 
   alias NervesHubLinkHTTP.{Client, HTTPFwupStream}
 
-  def update do
-    case Client.update() do
-      {:ok, %{"data" => %{"update_available" => true, "firmware_url" => url}}} ->
-        Logger.info("[NervesHubLinkHTTP] Downloading firmware: #{url}")
-        {:ok, http} = HTTPFwupStream.start_link(self())
-        # Spawn to allow async messages from FWUP.
-        spawn_monitor(HTTPFwupStream, :get, [http, url])
-        update_receive()
+  @doc """
+  Check for an available update.
+
+  The update is not automatically applied.
+  """
+  @spec update?(Keyword.t()) :: {:ok, String.t()} | {:ok, :no_update} | {:error, any()}
+  def update?(opts \\ []) do
+    case Client.update(opts) do
+      {:ok, %{"data" => %{"update_available" => true, "firmware_url" => url}}} -> {:ok, url}
+      {:ok, %{"data" => %{"update_available" => false}}} -> {:ok, :no_update}
+      {:error, _} = err -> err
+    end
+  end
 
-      {:ok, %{"data" => %{"update_available" => false}}} ->
-        :no_update
+  @doc """
+  Apply the update at the provided `url`.
+  """
+  @spec apply_update(String.t()) :: no_return()
+  def apply_update(url) when is_binary(url) do
+    Logger.info("[NervesHubLinkHTTP] Downloading firmware: #{url}")
+    {:ok, http} = HTTPFwupStream.start_link(self())
+    # Spawn to allow async messages from FWUP.
+    spawn_monitor(HTTPFwupStream, :get, [http, url])
+    update_receive()
+  end
 
-      {:error, _} = err ->
-        err
+  @doc """
+  Check for an available update and automatically apply it.
+  """
+  @spec update(Keyword.t()) :: no_return() | :no_update | {:error, any()}
+  def update(opts \\ []) do
+    case update?(opts) do
+      {:ok, :no_update} -> :no_update
+      {:ok, url} when is_binary(url) -> apply_update(url)
+      {:error, _} = err -> err
     end
   end
 
+  @doc """
+  Enter a receive loop handling messages from `NervesHubLinkHTTP.HTTPFwupStream`.
+  """
+  @spec update_receive() :: no_return()
   def update_receive() do
```

Comment:
I feel like maybe this should just be private? Is there any case we need to allow the caller access to this?

Reply:
I have never needed to call this manually myself. Happy to go either way here as to not hold up the other changes.

Reply:
Sweet. I feel like we can remove the doc/spec here and just make it private if you're good with making that change (even though its not _really_ what you were looking to change)